drivers/flash/flash_mspi_nor.h

```diff
@@ -115,6 +115,7 @@ struct flash_mspi_nor_data {
 	struct flash_mspi_nor_switch_info switch_info;
 	bool in_target_io_mode;
 	const struct mspi_dev_cfg *last_applied_cfg;
+	bool chip_detected;
```

Comment:
I think this name is misleading. The driver does not perform a chip detection, it just configures the chip according to what is specified in dts. I'd rather use `chip_initialized` or `chip_configured`.